cpp/src/interop/to_arrow.cu

```diff
@@ -295,8 +297,10 @@ std::shared_ptr<arrow::Array> dispatch_to_arrow::operator()<cudf::string_view>(
     return std::make_shared<arrow::StringArray>(
       0, std::move(tmp_offset_buffer), std::move(tmp_data_buffer));
   }
-  auto offset_buffer = child_arrays[0]->data()->buffers[1];
-  auto data_buffer   = child_arrays[1]->data()->buffers[1];
+  auto offset_buffer = child_arrays[strings_column_view::offsets_column_index]->data()->buffers[1];
+  // auto data_buffer   = child_arrays[1]->data()->buffers[1];
+  auto data_buffer =
+    fetch_data_buffer<int8_t>(strings_column_view{input_view}.chars(stream), ar_mr, stream);
```

Comment:
Could we have a `fetch_data_buffer` that accepts a pointer, size (e.g. `device_span`) so we don't need to create a `column_view` here?

Reply:
replaced with `device_span`